spring-batch-infrastructure/src/test/java/org/springframework/batch/item/file/builder/FlatFileItemReaderBuilderTests.java

```diff
@@ -273,6 +273,51 @@ public void testComments() throws Exception {
 		assertNull(reader.read());
 	}
 
+	@Test
+	public void testEmptyComments() throws Exception {
+		FlatFileItemReader<Foo> reader = new FlatFileItemReaderBuilder<Foo>()
+				.name("fooReader")
+				.resource(getResource("1,2,3\n4,5,6"))
+				.comments(new String[]{})
+				.delimited()
+				.names("first", "second", "third")
+				.targetType(Foo.class)
+				.build();
+
+		reader.open(new ExecutionContext());
+		Foo item = reader.read();
+		assertEquals(1, item.getFirst());
+		assertEquals(2, item.getSecond());
+		assertEquals("3", item.getThird());
+		item = reader.read();
+		assertEquals(4, item.getFirst());
+		assertEquals(5, item.getSecond());
+		assertEquals("6", item.getThird());
+		assertNull(reader.read());
+	}
+
+	@Test
+	public void testDefaultComments() throws Exception {
```

Comment:
Same here, to test default comment prefixes, I would use a resource having a line starting with `#` and assert that it is ignored by default:

```java
@Test
public void testDefaultComments() throws Exception {
	FlatFileItemReader<String> reader = new FlatFileItemReaderBuilder<String>()
			.name("itemReader")
			.resource(getResource("line1\n#line2"))
			.lineMapper(new PassThroughLineMapper())
			.build();

	reader.open(new ExecutionContext());
	String item = reader.read();
	assertEquals("line1", item);
	assertNull(reader.read()); // line "#line2" should be ignored by default
	reader.close();
}
```

However, this test fails not due to this PR, but due to another bug that I discovered while testing this PR, see [BATCH-2861](https://jira.spring.io/browse/BATCH-2861) and the suggested fix in #747. So please update the test and I will merge the PRs in the right order.

Reply:
The method `FlatFileItemReaderBuilder#comments` currently adds comments to the default empty list (it should have been called `addComments` IMO). I believe it should be rather:

```java
public FlatFileItemReaderBuilder<T> comments(String[] comments) {
	this.comments = Arrays.asList(comments);
	return this;
}
```

to be able to override the default comments (as described in BATCH-2837).

@drumonii Can you please add this change along with those in #747 here? Otherwise let me know if you agree and I will do it.